tasks/deploy/update.js

```diff
@@ -50,5 +54,30 @@ module.exports = function (gruntOrShipit) {
         shipit.log(chalk.green('Finished copy.'));
       });
     }
+
+    function setPreviousRevision() {
+      return shipit.getPreviousReleaseDirname().then(function(previousReleaseDir) {
+        if (previousReleaseDir) {
+          var file = path.join(shipit.releasesPath, previousReleaseDir, 'REVISION');
+          return shipit.remote('if [ -f ' + file + ' ]; then cat ' + file + ' 2>/dev/null; fi;').then(function(response) {
+
+            // TODO: How should we handle multiple?
+            var rev = response[0].stdout.trim();
+            shipit.previousRevision = rev || false;
+          });
+        }
+      });
+    }
+
+    function setCurrentRevision() {
+      shipit.log('Setting current revision and creating revision file.');
+
+      return shipit.local('git rev-parse ' + shipit.config.branch).then(function(response) {
```

Comment:
Just about coding style, for promise, put a break line before a new ".then", also try to avoid line with a length superior to ~ 200 chars (readability in editor and GitHub).